web/components/logentry/weight.jsx

```diff
@@ -0,0 +1,140 @@
+import CurrentEntryActions from '../../actions/current-entry-actions';
+import formUtils from '../../utils/form-utils';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TextBox from '../controls/text-box.jsx';
+
+import {
+	Col,
+	ControlLabel,
+	FormGroup,
+	Radio,
+	Row
+} from 'react-bootstrap';
+
+class Weight extends React.Component {
+	constructor() {
+		super();
+		this.onValueChanged = this.onValueChanged.bind(this);
+	}
+
+	onValueChanged(e) {
+		const weight = this.props.entry.weight
+			? Object.assign({}, this.props.entry.weight)
+			: {};
+
+		if (e.target.id) {
+			switch (e.target.id) {
```

Comment:
`switch` is redundant. Refactor this to just update weight and return if `e.target.id` is `'weight'`.